nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/StandardProcessorNode.java

```diff
@@ -1321,12 +1325,21 @@ public void run() {
                         this.unscheduled = true;
                     }
                     try {
-                        if (activeThreadMonitorCallback.call()) {
+                        if (scheduleState.isScheduled()) {
+                            schedulingAgent.unschedule(StandardProcessorNode.this, scheduleState);
+                        }
```

Comment:
Trying to understand. The @OnStopped is not invoked until unscheduled flag is set and given that OnUnschedule is a synchronous call how can ```if (scheduleState.isScheduled()) {``` ever be true if the internal 'unscheduled' flag is set.